File: server/medicore.apps/src/model/tagModel.ts
import { Type } from 'class-transformer';
import { ArticleTagMapModel } from './articleTagMapModel';
import { IEntityBase } from './IEntityBase';

export class TagModel implements IEntityBase {
    id: number;
    tagName: string;

    //region model
    @Type(() => ArticleTagMapModel)
    tagMaps: ArticleTagMapModel[];

    constructor() {
        this.id = 0;
        this.tagName = null;
        this.tagMaps = null;
    }
}

File: server/dna.core/dna.core.service/Models/TagModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace dna.core.service.Models
{
    public class TagModel
    {
        public int Id { get; set; }

        [Required, MaxLength(100)]
        public string TagName { get; set; }

        public virtual IEnumerable<ArticleTagMapModel> TagMaps { get; set; }
    }
}
